operator/controllers/resources/utils.go

```diff
@@ -94,3 +98,29 @@ func getReconciliationCondition(err error) metav1.Condition {
 
 	return condition
 }
+
+func getValidStructureCondition(err error) metav1.Condition {
```

Comment:
Maybe: getStructureConditionFromError?  I think this deserves a godoc explaining what is the error we are supposed to pass to it.